pymatgen/io/lammps/inputs.py

```diff
@@ -17,17 +17,293 @@
 from monty.dev import deprecated
 from monty.json import MSONable
 
+from pymatgen.io.core import InputFile
 from pymatgen.io.lammps.data import CombinedData, LammpsData
 from pymatgen.io.template import TemplateInputGen
 
-__author__ = "Kiran Mathew, Brandon Wood, Zhi Deng"
+__author__ = "Kiran Mathew, Brandon Wood, Zhi Deng, Manas Likhit, Guillaume Brunin"
 __copyright__ = "Copyright 2018, The Materials Virtual Lab"
 __version__ = "1.0"
 __maintainer__ = "Zhi Deng"
 __email__ = "z4deng@eng.ucsd.edu"
 __date__ = "Aug 1, 2018"
 
 
+class LammpsInputFile(InputFile):
+    """
+    Class representing a LAMMPS input settings file, e.g. in.lammps.
+    Allows for LAMMPS input generation in line/stage wise manner. A stage
+    here is defined as a block of LAMMPS input commands usually performing a
+    specific task during the simulation such as energy minimization or
+    NPT/NVT runs. But more broadly, a stage can also be block of LAMMPS
+    input where simulation box is setup, a set of variables are declared or
+    quantities are computed.
+    """
+
+    def __init__(self, input_settings: dict = None):
+        """
+        Args:
+            input_settings: dictionary of LAMMPS input settings.
+        """
+        self.nstages = 0
+        self.ncomments = 0
+        self.input_settings = input_settings if input_settings else {}
+
+    def add_stage(self, command: str | list | dict, stage_name: str = None):
+        """
+        Adds LAMMPS command(s) and its arguments to LAMMPS input file.
```

Comment:
Maybe a simple example for a stage here would help with understanding how this input file works.